worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala

```diff
@@ -175,7 +175,10 @@ private[deploy] class Controller(
     if (masterLocs.size() < requestMasterLocs.size()) {
       val msg = s"Not all master partition satisfied for $shuffleKey"
       logWarning(s"[handleReserveSlots] $msg, will destroy writers.")
-      masterLocs.asScala.foreach(_.asInstanceOf[WorkingPartition].getFileWriter.destroy())
+      masterLocs.asScala.foreach { partitionLocation =>
+        val fileWriter = partitionLocation.asInstanceOf[WorkingPartition].getFileWriter
+        fileWriter.destroy(new IOException(s"Reserve slots failed for FileWriter ${fileWriter}"))
```

Comment:
Here not means writer create failed, we should say
`Reserve slots failed for ${shuffleKey}` or even find out which writer failed to create